core/src/main/scala/zio/config/ConfigSource.scala

```diff
@@ -296,28 +305,53 @@ object ConfigSource {
     map: Map[String, A]
   )(f: A => ::[B], keyDelimiter: Option[Char], source: String): ConfigSource[String, B] =
     fromPropertyTrees(
-      unflatten(
-        map.map(
-          tuple => {
-            val vectorOfKeys = keyDelimiter match {
-              case Some(keyDelimiter) => tuple._1.split(keyDelimiter).toVector.filterNot(_.trim == "")
-              case None               => Vector(tuple._1)
-            }
-            vectorOfKeys -> f(tuple._2)
-          }
+      unwrapSingletonLists(
```

Comment:
What's the point of `unwrapSingletonLists` and `dropEmpty`?

Reply:
`fromMap` wraps every single value in `Sequence`, so there is no way to parse it as scalar.
There are also some tricky cases with empty `Sequence` in `fromMap` and `fromCommandLineArgs`, though there is no way to describe empty List in `Map[String, String]` or in command line arguments.

Reply:
Maybe `fromMap` should be changed so that it only wraps values in Sequence if there is more than one of them?

I agree on the problem of no way to describe empty lists that come in from a map. I'm not sure if we have a good solution to that, other than treating it specially when reading (e.g. allow reading an `Empty` to count as reading an empty list)